lock-passwordless/src/main/java/com/auth0/lock/passwordless/LockPasswordlessActivity.java

```diff
@@ -0,0 +1,443 @@
+
+    private static final String PASSWORDLESS_TYPE_PARAMETER = "PASSWORDLESS_TYPE_PARAMETER";
+    private static final String USERNAME_PARAMETER = "USERNAME_PARAMETER";
+    private static final String IS_IN_PROGRESS_PARAMETER = "IS_IN_PROGRESS_PARAMETER";
+
+    protected AuthenticationAPIClient client;
+    protected Bus bus;
+    Lock lock;
+    private int passwordlessType;
+    private boolean isInProgress;
+    private String username;
+    private LoginAuthenticationErrorBuilder errorBuilder;
+
+    public static void showFrom(Activity activity, int passwordlessType) {
+        Intent intent = new Intent(activity, LockPasswordlessActivity.class);
+        if (passwordlessType != MODE_EMAIL_CODE
+                && passwordlessType != MODE_EMAIL_MAGIC_LINK
+                && passwordlessType != MODE_SMS_CODE
+                && passwordlessType != MODE_SMS_MAGIC_LINK) {
+            Log.e(TAG, "Invalid passwordless type, it must be one of {MODE_EMAIL_CODE, MODE_EMAIL_MAGIC_LINK, MODE_SMS_CODE, MODE_SMS_MAGIC_LINK}");
+            return;
+        }
+        intent.putExtra(PASSWORDLESS_TYPE_PARAMETER, passwordlessType);
+        activity.startActivity(intent);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.com_auth0_passwordless_activity_lock_passwordless);
+
+        lock = getLock();
+        client = lock.getAuthenticationAPIClient();
+        bus = lock.getBus();
+
+        if (savedInstanceState == null) {
+            isInProgress = false;
+            Intent intent = getIntent();
+            setPasswordlessType(intent.getIntExtra(PASSWORDLESS_TYPE_PARAMETER, MODE_EMAIL_CODE));
+            AppLinkParser linkParser = new AppLinkParser();
+            boolean invalidMagicLink = linkParser.isAppLinkIntent(intent);
+            if (invalidMagicLink) {
+                setUseMagicLink(true);
+                int appLinkType = linkParser.getAppLinkTypeFromIntent(intent);
+                switch (appLinkType) {
+                    case AppLinkParser.TYPE_SMS:
+                        setIsEmailType(false);
+                        break;
+                    case AppLinkParser.TYPE_EMAIL:
+                        setIsEmailType(true);
+                        break;
+                    default:
+                        Log.e(TAG, "Started with an invalid app link intent:"+intent);
+                }
+            }
+
+            Fragment initialFragment;
+            if (isEmailType()) {
+                initialFragment = RequestCodeEmailFragment.newInstance(useMagicLink());
+            } else {
+                initialFragment = RequestCodeSmsFragment.newInstance(useMagicLink());
+            }
+            getSupportFragmentManager().beginTransaction()
+                    .add(R.id.com_auth0_container, initialFragment)
+                    .commit();
+
+            if (invalidMagicLink) {
```

Comment:
Shouldn't we add this fragment directly if the link is invalid and avoid replacing the "request code" fragment?